### PR TITLE
Final sync before version update

### LINKED ISSUE #26: Add components TableLoader, ColorPicker, ColorPalette
Part of #25

Add components 
- [ ] TableLoader
- [ ] ColorPicker
- [ ] ColorPalette

@yedhink _a

### LINKED ISSUE #55: [Syncing] formik/Select
## Points to note
- Component in action - `formik/Select.js`
- Github Urls to this component:
  * [neeto-ui/formik/Select.js](https://github.com/bigbinary/neeto-ui/tree/67ee4a5188c5feccbeca2767107651fb24932764/lib/components/formik/Select.js)
  * [NitroUI/Formik/Select.js](https://github.com/bigbinary/neeto-help-web/tree/0b11a7b0f98ac7c533df1e9ec37cf900b4cbc901/app/javascript/Admin/Components/NitroUI/Formik/Select.js)
- :green_circle: Green color signifies how that particular line is written in `neeto-ui` package
- :red_circle: Red color signifies how that particular line is written in `NitroUI` internals

## Sync Required or Not
- Are there significant changes for `formik/Select.js` that needs to be synced to `neeto-ui` package from `NitroUI`? - **No**
- This block seems unnecessary to me:
```diff
-            const targetAttr = field.name.split(".")[0];
-            if (Array.isArray(form.values[targetAttr])) {
-              formValues[targetAttr] = Object.values(formValues[targetAttr]);
-            }
-            setValues(formValues);
``` 
- @edwinbbu I need some clarification on whether to sync this part from `NitroUI` to `neetoui` or not. From my understanding, no need. The current `neetoui` version will do the job.

## Diff
```diff
--- NitroUI/formik/Select.js	2020-11-28 16:54:23.962073200 +0530
+++ neetoui-official/formik/Select.js	2020-11-28 16:49:18.771103400 +0530
@@ -21,7 +21,7 @@
         selectedValues.includes(optionValue(option))
       );
     } else {
-      return options.find(option => optionValue(option) === field.value) || "";
+      return options.find(option => optionValue(option) === field.value);
     }
   };
 
@@ -31,25 +31,16 @@
 
   return (
     <Field name={name}>
-      {({ field, form, meta }) => (
+      {({ field, form }) => (
         <Select
           options={options}
           name={field.name}
           value={options ? selectedOptions(options, field) : "Select an option"}
-          onChange={value => {
-            const formValues = R.set(
-              nestedLens(field.name),
-              value,
-              form.values
-            );
-            const targetAttr = field.name.split(".")[0];
-            if (Array.isArray(form.values[targetAttr])) {
-              formValues[targetAttr] = Object.values(formValues[targetAttr]);
-            }
-            setValues(formValues);
-          }}
-          onBlur={() => form.setFieldTouched(name, true)}
-          error={meta.touched && R.path(field.name.split("."), form.errors)}
+          onChange={value =>
+            setValues(R.set(nestedLens(field.name), value, form.values))
+          }
+          onBlur={field.onBlur}
+          error={form.errors[field.name]}
           getOptionValue={getOptionValue || (option => option.value)}
           isMulti={!!isMulti}
           {...rest}
```

### LINKED ISSUE #114: Final sync before version update
Some changes needs to be synced from NitroUI components excluding `form`.

This requires some manual work and time, since I have to go through the latest diffs once again.

@yedhink _a

### PR DESCRIPTION
**Pull request by yedhink**
Fixes #114 , Fixes #26, Fixes #55

I have synced the changes that happened in NitroUI once I completed the sync of those components to neeto-ui.

Have added the `chat.scss` which was referenced in `ColorPicker`.

I have avoided using this `Slugify` helper because I felt it was an anti-pattern. They should pass the sluggified value to the component rather than making the component do such side effects. [Refer this](https://github.com/bigbinary/neeto-help-web/blob/2bc0a146499644ef85f3a42572e1cb4102257a63/app/javascript/Admin/Components/NitroUI/Select.js#L45) to understand what i'm referring to.

I haven't added `Link` component since Edwin asked me not to.

I have synced 4 new components which was missed out in the process of syncing:
- ColorPalette
- ColorPicker
- TableLoader
- Accordion

The `formik/Select` had been modified since the previous one had some bugs. I had also cleaned up the code. I had covered the scenarios @dineshpanda had pointed out. @dineshpanda  please do let me know If I'm missing something. Thanks. Parent issues: 
- https://github.com/bigbinary/neeto-help-web/pull/9063
- https://github.com/bigbinary/neeto-help-web/pull/9073
- https://github.com/bigbinary/neeto-help-web/pull/9112

**From my understanding this is the last part of the syncing process, excluding the `form` files and folders. So do let me know if I have missed something or we need to make any more changes.**

Thanks and Cheers!

@edwinbbu _a

cc: @goutham-subramanyam  @vinay0x

### CODE REVIEW

File: lib/components/PageLoader.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import React from "react";
 import { BeatLoader } from "react-spinners";
 
-const PageLoader = ({ color = "#252f3f" }) => {
+const PageLoader = ({ color = "#252f3f", ...otherProps }) => {
   return (
-    <div className="relative w-full h-full">
+    <div className="relative w-full h-full" data-test-id="nitrohelp-page-loader" {...otherProps}>
```
Comment (goutham-subramanyam) [inline lib/components/PageLoader.js:6]: We don't need the default data test id.

File: lib/components/Switch.js [hunk 2]
```diff
@@ -46,20 +47,25 @@ const UncheckedState = ({ checked }) => (
...
+    className = "",
     onChange,
-    size,
     round = true,
     square = false,
     disabled = false,
-    label,
-    showStatus,
+    label = null,
+    showStatus = null,
```
Comment (goutham-subramanyam) [inline lib/components/Switch.js:57]: We can remove the showStatus block.

File: lib/components/Switch.js [hunk 3]
```diff
@@ -46,20 +47,25 @@ const UncheckedState = ({ checked }) => (
 const Switch = props => {
   const {
     name,
-    checked,
+    checked = false,
+    className = "",
     onChange,
-    size,
     round = true,
     square = false,
```
Comment (goutham-subramanyam) [inline lib/components/Switch.js:54]: These can be removed.

File: lib/components/Switch.js [hunk 4]
```diff
@@ -46,20 +47,25 @@ const UncheckedState = ({ checked }) => (
...
   return (
-    <div className="flex flex-row items-center justify-start">
-      <label className="flex m-0">
+    <div
+      className={classnames([
+        "flex flex-row items-center justify-start",
+        className
+      ])}
+    >
+      <Label className="flex m-0">
```
Comment (goutham-subramanyam) [inline lib/components/Switch.js:68]: Here it should be `<label` itself.

File: lib/components/Switch.js [hunk 5]
```diff
@@ -96,11 +103,11 @@ const Switch = props => {
...
             ) : (
-              <UncheckedState checked={checked} />
-            )}
+                <UncheckedState checked={checked} />
+              )}
           </span>
         </span>
-      </label>
+      </Label>
       {label && <span className="ml-2 switch__label">{label}</span>}
```
Comment (goutham-subramanyam) [inline lib/components/Switch.js:111]: We need to use `Label` here.

File: lib/layouts/PageHeading.js [hunk 6]
```diff
@@ -11,14 +11,15 @@ class PageHeading extends Component {
       breadcrumbLinks = null,
```
Comment (goutham-subramanyam) [inline lib/layouts/PageHeading.js:11]: We don't need to sync `PageHeading`. We should be syncing `Header` from layouts.

File: lib/shared/OutsideClickHandler.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import React, { useRef, useEffect } from "react";
```
Comment (goutham-subramanyam) [inline lib/shared/OutsideClickHandler.js:1]: We don't need this component.

File: lib/styles/index.scss [hunk 8]
```diff
@@ -27,5 +27,8 @@
 @import "./layout/sidebar";
 @import "./layout/header";
 
+//Screens
+@import "./screens/chat.scss";
```
Comment (goutham-subramanyam) [inline lib/styles/index.scss:31]: No need to import this file.

File: src/index.js [hunk 9]
```diff
@@ -181,6 +194,27 @@ const App = () => {
...
+      />
+
+      <ColorPicker
+        handleChange={value => { }}
+        color="#fefefe"
+        active={true}
+      />
+
+      <Accordion className="mt-2" defaultActiveKey={0}>
+        <Accordion.Item title="Opened Heading" isOpen>
```
Comment (goutham-subramanyam) [inline src/index.js:210]: No need to pass `isOpen`. It is handled internally.

**Review by goutham-subramanyam — CHANGES_REQUESTED** [review]
@yedhink _a

File: lib/components/Switch.js (hunk 5 above)
Comment (yedhink) [reply lib/components/Switch.js:111]: @goutham-subramanyam 

Just to be more clear, you meant to replace `span` with `Label`, correct?

Comment (goutham-subramanyam) [reply lib/components/Switch.js:111]: Yes

File: src/index.js (hunk 9 above)
Comment (yedhink) [reply src/index.js:210]: This is an example an item can be opened by default.

File: lib/layouts/PageHeading.js (hunk 6 above)
Comment (edwinbbu) [reply lib/layouts/PageHeading.js:11]: @yedhink Let's handle layouts separately and not on this PR.

cc @goutham-subramanyam

**Comment by yedhink** [comment]
@goutham-subramanyam _a

Please review.

File: lib/components/Switch.js [hunk 10]
```diff
@@ -96,15 +100,12 @@ const Switch = props => {
...
+                <UncheckedState checked={checked} />
+              )}
           </span>
         </span>
       </label>
-      {label && <span className="ml-2 switch__label">{label}</span>}
-      {showStatus && (
-        <span className="switch__label">{checked ? "Active" : "Inactive"}</span>
-      )}
+      {label && <Label className="ml-2 switch__label">{label}</Label>}
```
Comment (goutham-subramanyam) [inline lib/components/Switch.js:108]:
```suggestion
      {label && <Label className="ml-3">{label}</Label>}
```

File: src/index.js [hunk 11]
```diff
@@ -56,6 +59,16 @@ const App = () => {
     }
   ];
 
+  const DEFAULT_COLORS = [
```
Comment (goutham-subramanyam) [inline src/index.js:62]: This is not needed here anymore right ?

**Review by goutham-subramanyam — COMMENTED** [review]
@yedhink _a

**Comment by yedhink** [comment]
@goutham-subramanyam _a

Please review.

**Review by goutham-subramanyam — APPROVED** [review]
LGTM. Merging.

**Pull request merged**